corehq/apps/geospatial/static/geospatial/js/case_grouping_map.js

```diff
@@ -280,15 +280,19 @@ hqDefine("geospatial/js/case_grouping_map",[
         setMapLayersVisibility(MAPBOX_LAYER_VISIBILITY.None);
         mapMarkers.forEach((marker) => marker.remove());
         mapMarkers = [];
+
+        let groupColorByID = _.object(_.map(caseGroupsInstance.generatedGroups, function(group) {
+            return [group.groupId, group.color]
+        }));
+
         exportModelInstance.casesToExport().forEach(function (caseItem) {
             const coordinates = caseItem.itemData.coordinates;
             if (!coordinates) {
                 return;
             }
             const caseGroupID = caseItem.groupId;
             if (caseGroupsInstance.groupIDInVisibleGroupIds(caseGroupID)) {
-                let caseGroup = caseGroupsInstance.getGroupByID(caseGroupID);
-                color = caseGroup.color;
+                color = groupColorByID[caseGroupID];
```

Comment:
`getGroupByID` iterates over groups to finding the matching group each time its called. That is avoided by simply keeping a mapping of group ID to color once.